src/internal/m365/collection/site/backup.go

```diff
@@ -185,17 +188,33 @@ func CollectLists(
 			el.AddRecoverable(ctx, clues.WrapWC(ctx, err, "creating list collection path"))
 		}
 
-		collection := NewPrefetchCollection(
-			bh,
-			dir,
-			ac,
-			scope,
-			su,
-			bpc.Options)
-
-		collection.AddItem(
-			ptr.Val(list.GetId()),
-			ptr.Val(lmt))
+		if lmt.IsZero() {
+			prefetchCol := NewPrefetchCollection(
+				bh,
+				dir,
+				ac,
+				scope,
+				su,
+				bpc.Options)
+
+			prefetchCol.AddItem(
+				ptr.Val(list.GetId()),
+				ptr.Val(lmt))
+
+			collection = prefetchCol
+		} else {
```

Comment:
nit: A pattern we follow throughout the codebase is that instead of if else, we initialize with a certain type e.g. `collection := NewLazyFetchCollection{...`. And then we do `if condition { collection := NewPrefetchCollection{...`

Reply:
Definitely. In this case `AddItem` method is tied to both collection separately and cannot be called on the `collection` var. 

Let me make a change to avoid the if-else.